src/brogue/IO.c

```diff
@@ -3482,10 +3482,7 @@ void refreshSideBar(short focusX, short focusY, boolean focusedEntityMustGoFirst
 
     // Initialization.
     displayEntityCount = 0;
-    for (i=0; i<ROWS*2; i++) {
-        rogue.sidebarLocationList[i][0] = -1;
-        rogue.sidebarLocationList[i][1] = -1;
-    }
+    memset(rogue.sidebarLocationList, -1, sizeof(rogue.sidebarLocationList));
```

Comment:
It works as intended here because `memset` converts -1 (int) to unsigned char, which is `0xff`, and all signed integer types, including short, encode -1 as a sequence of `0xff`. It wouldn't work for e.g -2, because -2 (int) is `ff fe` which `memset` converts to `fe` and `fe fe` as a short is -258, not -2. It would not work for most values above 255 either.

I measured with `DEBUG=NO`, correct, but to find the exact lines that were consuming the most time within that hot function I had to trust `gcov`. I tuned the lines that were executing the highest number of times, and these 4 lines were part of it.

Reply:
I don't know if I'm being overly C pedantic but I'm not keen; besides number 0, signed representation is implementation-defined.. so this is not a very well-defined operation

Reply:
Thanks, that's good to know! I'm reverting it to a well-behaved loop. Performance remains pretty much the same without this micro-optimization anyway.